src/main/scala/ch/epfl/bluebrain/nexus/kg/async/Projects.scala

```diff
@@ -41,7 +45,40 @@ trait Projects[F[_]] {
     * @param ref the project reference
     * @return the collection of known resolvers configured for the argument project
     */
-  def revolvers(ref: ProjectRef): F[Set[Resolver]]
+  def resolvers(ref: ProjectRef): F[Set[Resolver]]
+
+  /**
+    * Adds the resolver to the collection of project resolvers.
+    *
+    * @param ref       the project reference
+    * @param resolver  the resolver to add
+    * @param updateRev whether to update the resolver collection if the resolver provided has a higher revision than an
+    *                  already existing element in the collection with the same id
+    * @return true if the update was performed or false if the element was already in the set
+    */
+  def addResolver(ref: ProjectRef, resolver: Resolver, updateRev: Boolean): F[Boolean]
+
+  /**
+    * Removes the resolver identified by the argument id from the collection of project resolvers.
+    *
+    * @param ref       the project reference
+    * @param id        the id of the resolver to remove
+    * @param timestamp the instant used to merge the register value
```

Comment:
no special reason, it was just the type required by the `TimestampedValue[A]`, but it may be easier to just use it since we deal with instants in the resource types.

Reply:
this made me realize that we can't use the resource instant, but we must pass explicitly the instant of the event, otherwise merges won't happen correctly across the cluster.